homeassistant/components/shelly/__init__.py

```diff
@@ -233,6 +240,13 @@ async def _async_setup_rpc_entry(hass: HomeAssistant, entry: ShellyConfigEntry)
         port=get_http_port(entry.data),
     )
 
+    if entry.data.get(CONF_MODEL) == MODEL_WALL_DISPLAY and not entry.data.get(
+        CONF_SCRIPT
+    ):
+        data = {**entry.data}
+        data[CONF_SCRIPT] = False
+        hass.config_entries.async_update_entry(entry, data=data)
+
```

Comment:
This will happen only after the config flow has already evaluated the condition if options are allowed or not and is not effective until device is reloaded, I also think it just complicates the PR, PR was already ready with a very simple solution, why do we need to complicate it?
```suggestion
```

Reply:
I think that if we store for other devices the info of `script=False` and we know a device should be listed, we should update it.

The best option is indeed to migrate the config entry.
But I applied the same we did for the wrong sleep period.

Reply:
but as I explained it won't work for the first time as the device will load with the older value, this also makes a simple PR complicated, I already tested and validated a very simple solution (which you created) and now we complicate it. It would also mean we need to add tests for the migration of the data